site/src/content/docs/ref/package-values.mdx

```diff
@@ -0,0 +1,427 @@
+
+```yaml
+components:
+  - name: my-component
+    manifests:
+      - name: app-manifests
+        template: true  # Enables Go template processing
+        files:
+          - deployment.yaml
+          - service.yaml
+          - configmap.yaml
+```
+
+See the [Templating Reference](/ref/templating/) for complete details on template syntax, Sprig functions, and advanced usage.
+
+### In Actions
+
+Actions require explicit opt-in to templating. Set `template: true` on individual actions:
+
+```yaml
+components:
+  - name: my-component
+    actions:
+      onCreate:
+        after:
+          - cmd: |
+              echo "Deploying {{ .Values.site.name }}"
+              echo "Environment: {{ .Values.app.environment }}"
+            template: true  # Required to enable templating in actions
+```
+
+See the [Templating Reference](/ref/templating/) for details on templating in actions and why it requires opt-in.
+
+#### Setting Values from Actions
+
+Actions can dynamically set values using the `setValues` field, similar to `setVariables`:
+
+```yaml
+components:
+  - name: dynamic-config
+    actions:
+      onDeploy:
+        after:
+          - cmd: echo "generated-password-123"
+            setValues:
+              - name: database.password
+
+          - cmd: kubectl get configmap app-config -o json
+            setValues:
+              - name: existing.config
+                type: json
+
+          - cmd: |
+              cat <<EOF
+              apiVersion: v1
+              kind: Config
+              settings:
+                debug: true
+              EOF
+            setValues:
+              - name: runtime.settings
+                type: yaml
+```
+
+The `setValues` field accepts:
+- `name`: The path where the value should be stored (e.g., `database.password`)
+- `type`: The format of the output - `string` (default), `json`, or `yaml`
+
+Values set via `setValues` are available in subsequent actions and manifests within the same deployment.
+
+### Mapping to Helm Chart Values
+
+Map Package Values to Helm chart values using `sourcePath` and `targetPath`:
+
+```yaml
+components:
+  - name: helm-component
+    charts:
+      - name: my-chart
+        version: 1.0.0
+        namespace: my-app
+        localPath: charts/my-chart
+        values:
+          # Map .Values.app.name to .appName in the Helm chart
+          - sourcePath: ".app.name"
+            targetPath: ".appName"
+
+          # Map nested values
+          - sourcePath: ".app.replicas"
+            targetPath: ".replicaCount"
+
+          # Map to deeply nested chart values
+          - sourcePath: ".database.host"
+            targetPath: ".config.database.host"
+
+          # Multiple mappings to the same target - last one wins
+          - sourcePath: ".database.host"
+            targetPath: ".config.database.host"
+          - sourcePath: ".database.host_override"
+            targetPath: ".config.database.host"  # Takes priority
+```
+
+#### Mapping All Values
```

Comment:
nit: we may want to refine this section at some point to help users understand the best practices for building a package with values files that aligns to broader zarf hardening goals.

said another way - I think we should encourage using sourcePath/targetPath as a best practice for package hardening.